src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp

```diff
@@ -0,0 +1,429 @@
+    SAY_THALORIEN_3 = 2,
+    SAY_THALORIEN_4 = 3,
+    SAY_THALORIEN_5 = 4,
+    SAY_THALORIEN_6 = 5,
+    SAY_THALORIEN_7 = 6,
+    SAY_THALORIEN_8 = 7,
+    SAY_THALORIEN_9 = 8,
+    SAY_THALORIEN_10 = 9,
+
+    SAY_MORLEN_1 = 0,
+    SAY_MORLEN_2 = 1,
+    SAY_MORLEN_3 = 2,
+    SAY_MORLEN_4 = 3,
+    SAY_MORLEN_5 = 4
+};
+
+enum ThalorienDawnseekerSummons
+{
+    SUMMON_SUNWELL_DEFENDER = 1,
+    SUMMON_SCOURGE_ZOMBIE   = 2,
+    SUMMON_GHOUL_INVADER    = 3,
+    SUMMON_CRYPT_RAIDER     = 4
+};
+
+enum ThalorienDawnseekerActions
+{
+    ACTION_START_QUEST = 1
+};
+
+enum ThalorienDawnseekerEvents
+{
+    EVENT_INTRO_0           = 1,
+    EVENT_INTRO_1           = 2,
+    EVENT_INTRO_2           = 3,
+    EVENT_INTRO_3           = 4,
+    EVENT_SALUTE            = 5,
+    EVENT_DEFENDERS_RUN     = 6,
+    EVENT_DEFENDERS_RUN_2   = 7,
+    EVENT_THALORIEN_GO      = 8,
+    EVENT_INTRO_4           = 9,
+    EVENT_SUMMON_MORLEN     = 10,
+    EVENT_INTRO_5           = 11,
+    EVENT_MORLEN_1          = 12,
+    EVENT_SPAWN_WAVE_1      = 13,
+    EVENT_SPAWN_WAVE_2      = 14,
+    EVENT_SPAWN_WAVE_3      = 15,
+    EVENT_WAVE_ATTACK       = 16,
+    EVENT_MORLEN_2          = 17,
+    EVENT_MORLEN_ATTACK     = 18,
+    EVENT_OUTRO_1           = 19,
+    EVENT_OUTRO_2           = 20,
+    EVENT_OUTRO_3           = 21,
+    EVENT_OUTRO_4           = 22,
+    EVENT_KNEEL             = 23
+};
+
+enum ThalorienDawnseeker
+{
+    MENU_EXAMINE_REMAINS        = 10908,
+
+    NPC_THALORIEN_DAWNSEEKER    = 37205,
+    NPC_SUNWELL_DEFENDER        = 37211,
+    NPC_MORLEN_GOLDGRIP         = 37542,
+    NPC_SCOURGE_ZOMBIE          = 37538,
+    NPC_GHOUL_INVADER           = 37539,
+    NPC_CRYPT_RAIDER            = 37541,
+
+    SPELL_BLOOD_PRESENCE        = 50689,
+    SPELL_KILL_CREDIT           = 70265,
+    SPELL_POLYMORPH_VISUAL      = 27123
+};
+
+float const defenderReverse = 5.67232f;
+Position const defenderRun[10] =
+        {
+                { 11934.78f, -7064.424f, 33.38046f },
+                { 11927.9f, -7061.303f, 32.00986f },
+                { 11927.81f, -7067.289f, 31.81058f },
+                { 11927.89f, -7058.299f, 32.13486f },
+                { 11927.9f, -7061.303f, 32.00986f },
+
+                { 11934.78f, -7064.424f, 33.38046f },
+                { 11920.69f, -7061.134f, 30.40684f },
+                { 11920.56f, -7067.133f, 30.41605f },
+                { 11920.76f, -7058.135f, 30.61656f },
+                { 11920.5f, -7070.132f, 30.40836f }
+        };
+
+Position const thalorienSummon = { 11795.32f, -7070.476f, 26.27511f, 5.67232f  };
+Position const thalorienFight = { 11788.46f, -7063.375f, 25.79677f, 3.054326f };
+Position const morlenSummon = { 11766.46f, -7050.078f, 26.19846f, 5.637414f  };
+
+class npc_thalorien_dawnseeker : public CreatureScript
+{
+public:
+    npc_thalorien_dawnseeker() : CreatureScript("npc_thalorien_dawnseeker") { }
+
+    struct npc_thalorien_dawnseekerAI : public ScriptedAI
```

Comment:
You can make your code shorter by using new script register thingy. Lookup RegisterCreatureAI examples :)